### PR TITLE
[DOCS] Adds descriptions to index APIs (G-Z)

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

Related to https://github.com/elastic/elasticsearch-specification/issues/964.

Adds descriptions to index API properties (G-Z). Source of info: https://www.elastic.co/guide/en/elasticsearch/reference/master/indices.html

### CODE REVIEW

File: specification/indices/put_data_lifecycle/IndicesPutDataLifecycleRequest.ts [hunk 1]
```diff
@@ -29,14 +29,41 @@ import { Duration } from '@_types/Time'
...
+     * Supports wildcards (`*`).
+     * To target all data streams use `*` or `_all`.
+     */
     name: DataStreamNames
   }
   query_parameters: {
+    /**
+     * Type of data stream that wildcard patterns can match.
+     * Supports comma-separated values, such as `open,hidden`.
+     * Valid values are: all, hidden, open, closed, none.
```
Comment (abdonpijpelink) [inline specification/indices/put_data_lifecycle/IndicesPutDataLifecycleRequest.ts:43]:
```suggestion
     * Valid values are: `all`, `hidden`, `open`, `closed`, `none`.
```

File: specification/indices/resolve_index/ResolveIndexRequest.ts [hunk 2]
```diff
@@ -21,15 +21,28 @@ import { RequestBase } from '@_types/Base'
...
+ * Multiple patterns and remote clusters are supported.
  * @rest_spec_name indices.resolve_index
  * @availability stack since=7.9.0 stability=stable
  * @availability serverless stability=stable visibility=public
  */
 export interface Request extends RequestBase {
   path_parts: {
+    /**
+     * Comma-separated name(s) or index pattern(s) of the indices, aliases, and data streams to resolve
+     *  Resources on remote clusters can be specified using the` <cluster>`:`<name>` syntax.
```
Comment (abdonpijpelink) [inline specification/indices/resolve_index/ResolveIndexRequest.ts:34]:
```suggestion
     * Comma-separated name(s) or index pattern(s) of the indices, aliases, and data streams to resolve.
     * Resources on remote clusters can be specified using the `<cluster>`:`<name>` syntax.
```

**Review by abdonpijpelink — APPROVED** [review]
LGTM. I've left a few suggestions for minor typos.

**Comment by szabosteve** [comment]
Thank you!

**Pull request merged**